firmware/library/L1_Drivers/esp8266.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+#include "config.hpp"
+
+class Esp8266WifiInterface
+{
+ public:
+    std::array<char, config::kEsp8266BufferSize> default_buffer;
+    enum class ConnectionType : uint8_t
+    {
+        kTcp = 0,
+        kUdp = 1
+    };
+    virtual bool InitializeClientMode()                          = 0;
+    virtual bool TestModule()                                    = 0;
+    virtual void ResetModule()                                   = 0;
+    virtual void DeepSleep(uint32_t time)                        = 0;
+    virtual void GetModuleVersion()                              = 0;
+    virtual void GetNetworkConnection()                          = 0;
+    virtual bool ConnectToAccessPoint(
+        const char ssid[], const char password[])                = 0;
+    virtual void DisconnectFromAccessPoint()                     = 0;
+    virtual bool IsConnectedToAccessPoint()                      = 0;
+    virtual bool ConnectToRemoteIp(
+        const char remote_ip[], const char port[],
```

Comment:
I'd rather this say hostname rather than ip address. Since either one can be used, I think we should favor a variable name that fits closer to what people are going to remember.

I can remember google.com, yahoo.com, sjsu.edu, but I cannot remember or track their ip addresses.